common/http.go

```diff
@@ -37,3 +39,35 @@ func InterpretError(res *http.Response, body []byte) error {
 
 	return NewHTTPStatusError(res.StatusCode, fmt.Errorf("%w: %s", ErrUnknown, string(body)))
 }
+
+func getURL(baseURL string, urlString string) (string, error) {
+	if strings.HasPrefix(urlString, "http://") || strings.HasPrefix(urlString, "https://") {
+		return urlString, nil
+	}
+
+	if len(baseURL) == 0 {
+		return "", fmt.Errorf("%w (input is %q)", ErrEmptyBaseURL, urlString)
+	}
+
+	return url.JoinPath(baseURL, urlString)
+}
+
+// ErrorHandler allows the caller to inject their own HTTP error handling logic.
+// All non-2xx responses will be passed to the error handler. If the error handler
+// returns nil, then the error is ignored and the caller is responsible for handling
+// the error. If the error handler returns an error, then that error is returned
+// to the caller, as-is. Both the response and the response body are passed
+// to the error handler as arguments.
+type ErrorHandler func(rsp *http.Response, body []byte) error
+
+// HTTPClient is an HTTP client which makes certain assumptions, such as
+// that the response body is JSON. It also handles OAuth access token refreshes.
+type HTTPClient struct {
```

Comment:
Since now the HTTPClient is not just JSONHTTPClient, renamed and moved to this file from json.go